libs/utils/android/workload.py

```diff
@@ -88,6 +88,29 @@ def run(self, out_dir, collect='',
             **kwargs):
         raise RuntimeError('Not implemeted')
 
+    def setupTargetScript(self, actions):
+        """
+        Push a script to the target
+
+        :param actions: List of actions(commands) to run
+        :type actions: list(str)
+        """
+
+        actions = ['#!{} sh'.format(self._target.busybox)] + actions
+        actions = str.join('\n', actions)
+
+        self._scriptpath = '{}/remote_script.sh'.format(self._target.executables_directory)
+
+        self._target.execute('echo \"{}\" > {}'.format(actions, self._scriptpath))
```

Comment:
I agree, we should build a script locally and push on target for execution. This allows also to keep a local copy of the executed script in the `Workload::run::out_dir` results folder.